app/models/target.rb

```diff
@@ -1,5 +1,52 @@
 class Target < ApplicationRecord
+  acts_as_mappable :default_units => :kms, :lat_column_name => :latitude, :lng_column_name => :longitude
   belongs_to :user
   has_one :topic
 
+  after_commit :search_for_compatible_targets, on: [:create, :update]
+
+  private
+
+  def search_for_compatible_targets
+
+    origin = Geokit::LatLng.new(self.latitude, self.longitude)
+    compatibleTargets = Target.within(self.radius * 0.001, :units => :kms, :origin => origin).all.select { |t| t.user_id != self.user_id && t.topic_id == self.topic_id }
```

Comment:
can this be a scope or extract `|t| t.user_id != self.user_id && t.topic_id == self.topic_id` to a method with a proper name